test/unit/json_schema/custom_vocabularies_test.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:json_schema/json_schema.dart';
+import 'package:json_schema/src/json_schema/format_exceptions.dart';
+import 'package:json_schema/src/json_schema/type_validators.dart';
+import 'package:test/test.dart';
+
+main() {
+  var ck = CustomVocabulary(
+    Uri.parse("http://localhost:4321/vocab/min-date"),
+    {"minDate": KeywordProcessor(_minDateSetter, _validateMinDate)},
+  );
+  group("Custom Vocabulary Tests", () {
+    test('Should process custom vocabularies and validate', () async {
+      final schema = await JsonSchema.createAsync(
+        {
+          r'$schema': 'http://localhost:4321/date-keyword-meta-schema.json',
+          r'$id': 'http://localhost:4321/date-keword-schema',
+          'properties': {
+            "publishedOn": {"minDate": "2020-12-01"},
+            "baz": {"type": "string"}
+          },
+          "required": ["baz", "publishedOn"]
+        },
+        schemaVersion: SchemaVersion.draft2020_12,
+        customKeyword: ck,
```

Comment:
What if you want to specify multiple custom vocabs?

Reply:
I always intended to pass i list in here, I just didn't get that done before pushing this branch up.